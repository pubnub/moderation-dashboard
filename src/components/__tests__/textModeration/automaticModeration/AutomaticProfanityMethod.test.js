import React from "react";
import { shallow } from "enzyme";
import AutomaticProfanityMethod from "../../../textModeration/automaticModeration/AutomaticProfanityMethod";
import { mockTextModerationProps } from "../../../mockTest/mockTextProfanity";

const expectTextValue = (a, b) => {
  expect(a.text()).toBe(b);
};

describe("Automatic Text Moderation", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(
      <AutomaticProfanityMethod state={mockTextModerationProps()} setState={jest.fn()} />
    );
  });

  test("Snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Check Labels", () => {
    expectTextValue(wrapper.find('[testId="Channel_Id"]'), "Channel ID");
    expectTextValue(wrapper.find('[testId="applyAll_channel"]'), "Apply to All Channel IDs");
    expectTextValue(
      wrapper.find('[testId="SelectToolLabel"]'),
      "Select third party tool for Automatic Detection"
    );
    expectTextValue(wrapper.find('[testid="tisaneApiKey"]'), "Tisane.ai API Key");
    expectTextValue(wrapper.find('[testid="lang"]'), "Language");
    expectTextValue(
      wrapper.find('[testid="BigotryLevel"]'),
      "Filter Level For Bigotry (Hate Speech)"
    );
    expectTextValue(
      wrapper.find('[testid="CyberBullying"]'),
      "Filter Level For Personal Attacks (Cyberbullying)"
    );
    expectTextValue(wrapper.find('[testid="criminalLevel"]'), "Filter Level For Criminal Activity");
    expectTextValue(wrapper.find('[testid="sexualLevel"]'), "Filter Level For Sexual Advances");
    expectTextValue(wrapper.find('[testid="ProfanityLevel"]'), "Filter Level For Profanity");
    expectTextValue(wrapper.find('[testid="blockOrMask"]'), "When profanity is detected");
    expectTextValue(wrapper.find('[testid="setChar"]'), "Set a masking character");
    expectTextValue(wrapper.find('[testid="reRouteMsg"]'), "Route messages to");
  });

  test("Check initial Values", () => {
    expect(wrapper.find("#channelId").getElement().props.value).toBe("*");
    expect(wrapper.find("#selectTool").getElement().props.value).toBe("tisane");
    expect(wrapper.find("#tisaneApiKey").getElement().props.value).toBe("");
    expect(wrapper.find("#lang").getElement().props.value).toBe("Hindi");
    expect(wrapper.find("#BigotryLevel").getElement().props.value).toBe(0.5);
    expect(wrapper.find("#CyberBullying").getElement().props.value).toBe(0.5);
    expect(wrapper.find("#criminalLevel").getElement().props.value).toBe(0.5);
    expect(wrapper.find("#sexualLevel").getElement().props.value).toBe(0.5);
    expect(wrapper.find("#ProfanityLevel").getElement().props.value).toBe(0.5);
    expect(wrapper.find("#blockOrMask").getElement().props.value).toBe("mask-message");
    expect(wrapper.find("#setChar").getElement().props.value).toBe("*");
  });

  test("Check Placeholders", () => {
    expect(wrapper.find("#channelId").getElement().props.placeholder).toBe(
      "Enter channel or channel pattern here Under18.*, Under18"
    );
    expect(wrapper.find("#tisaneApiKey").getElement().props.placeholder).toBe("Tisane.ai API key");
    expect(wrapper.find("#setChar").getElement().props.placeholder).toBe("Set");
  });

  test("Tisane Handler", () => {
    const LangChange = wrapper.find("#lang").simulate("change", {
      target: { name: "tisaneLanguage", value: "Spanish" },
    });
    expect(LangChange).toBeTruthy();

    const KeyChange = wrapper.find("#tisaneApiKey").simulate("change", {
      target: { name: "tisaneApiKey", value: "newKey" },
    });
    expect(KeyChange).toBeTruthy();

    const CyberChange = wrapper
      .find("#CyberBullying")
      .simulate("change", { target: { value: "" } }, "newCyberValue");
    expect(CyberChange).toBeTruthy();

    const CriminalChange = wrapper
      .find("#criminalLevel")
      .simulate("change", { target: { value: "" } }, "newCriminalValue");
    expect(CriminalChange).toBeTruthy();

    const BigotryChange = wrapper
      .find("#BigotryLevel")
      .simulate("change", { target: { value: "" } }, "newBiogotry");
    expect(BigotryChange).toBeTruthy();

    const SexualChange = wrapper
      .find("#sexualLevel")
      .simulate("change", { target: { value: "" } }, "newSexual");
    expect(SexualChange).toBeTruthy();

    const ProfanityChange = wrapper
      .find("#ProfanityLevel")
      .simulate("change", { target: { value: "" } }, "newProfanityValue");
    expect(ProfanityChange).toBeTruthy();
  });

  test("Check CheckBoxes", () => {
    expect(wrapper.find("#applyAll").getElement().props.checked).toBe(true);
    expect(wrapper.find("#reRoute").getElement().props.checked).toBe(false);
  });

  test("Check Save Button", () => {
    expect(wrapper.find("#save").text()).toBe("Save");
  });

  test("SiftNinja Handler", () => {
    wrapper = shallow(
      <AutomaticProfanityMethod state={mockTextModerationProps("siftninja")} setState={jest.fn()} />
    );

    const ToolChange = wrapper.find("#selectTool").simulate("change", {
      target: { name: "toolForAutomaticDetection", value: "tisane" },
    });
    expect(ToolChange).toBeTruthy();

    const ChannelChange = wrapper.find("#channelId").simulate("change", {
      target: { name: "automaticDetectionChannel", value: "tisane" },
    });
    expect(ChannelChange).toBeTruthy();

    const ReRouteChange = wrapper.find("#reRoute").simulate("change", {
      target: { name: "automaticDetectionReRouteMessages", checked: false },
    });
    expect(ReRouteChange).toBeTruthy();

    const ModChange = wrapper.find("#blockOrMask").simulate("change", {
      target: { name: "automaticDetectionModType", value: "newModType" },
    });
    expect(ModChange).toBeTruthy();

    const VulgarChange = wrapper
      .find("#vulgarRiskFactor")
      .simulate("change", { target: { value: "" } }, "newVulgar");
    expect(VulgarChange).toBeTruthy();

    const SextingChange = wrapper
      .find("#SextingRisk")
      .simulate("change", { target: { value: "" } }, "newSextingValue");
    expect(SextingChange).toBeTruthy();

    const RacismChange = wrapper
      .find("#racismRisk")
      .simulate("change", { target: { value: "" } }, "newRacismValue");
    expect(RacismChange).toBeTruthy();

    const AccountChange = wrapper.find("#siftAccountName").simulate("change", {
      target: { name: "siftNinjaAccountName", value: "newAccount" },
    });
    expect(AccountChange).toBeTruthy();

    const SiftChannelChange = wrapper.find("#siftChannelName").simulate("change", {
      target: { name: "siftNinjaChannelName", value: "newAccount" },
    });
    expect(SiftChannelChange).toBeTruthy();

    const ApiKeyChange = wrapper.find("#siftApiKey").simulate("change", {
      target: { name: "siftNinjaApiKey", value: "newKey" },
    });
    expect(ApiKeyChange).toBeTruthy();

    const CharChange = wrapper.find("#setChar").simulate("change", {
      target: { name: "automaticDetectionCharacterToMaskWith", value: "@" },
    });
    expect(CharChange).toBeTruthy();

    const ApplyAllChannelTrue = wrapper
      .find("#applyAll")
      .simulate("change", { target: { checked: true } });
    expect(ApplyAllChannelTrue).toBeTruthy();

    const ApplyAllChannelFalse = wrapper
      .find("#applyAll")
      .simulate("change", { target: { checked: false } });
    expect(ApplyAllChannelFalse).toBeTruthy();
  });
});
