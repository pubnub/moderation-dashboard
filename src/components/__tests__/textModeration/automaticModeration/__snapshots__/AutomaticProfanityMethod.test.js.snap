// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Automatic Text Moderation Snapshot 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Card))
    className="makeStyles-cardRoot-3"
  >
    <Styled(MuiBox)>
      <WithStyles(ForwardRef(Grid))
        container={true}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <label
            className="makeStyles-label-2"
            testId="Channel_Id"
          >
            Channel ID
          </label>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <Styled(MuiBox)
            className="makeStyles-infoIcon-9"
          >
            <WithStyles(WithStyles(ForwardRef(Tooltip)))
              title="Note: Please add a Channel ID or a Channel pattern. eg. channel.* OR pubNub. Channel ID pattern applies to BOTH text & image moderation if both are enabled."
            >
              <img
                alt="info-circle"
                src="/images/info-circle.svg"
              />
            </WithStyles(WithStyles(ForwardRef(Tooltip)))>
          </Styled(MuiBox)>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
      <Styled(MuiBox)
        mb={2}
        pt={1}
      >
        <WithStyles(ForwardRef(TextField))
          disabled={true}
          error={false}
          fullWidth={true}
          id="channelId"
          name="automaticDetectionChannel"
          onChange={[Function]}
          placeholder="Enter channel or channel pattern here Under18.*, Under18"
          size="small"
          value="*"
          variant="outlined"
        />
      </Styled(MuiBox)>
    </Styled(MuiBox)>
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <WithStyles(ForwardRef(Grid))
        container={true}
        justify="flex-start"
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <WithStyles(ForwardRef(Checkbox))
            checked={true}
            id="applyAll"
            onChange={[Function]}
          />
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <Styled(MuiBox)
            pl={1}
            pt={1}
          >
            <WithStyles(ForwardRef(Typography))
              testId="applyAll_channel"
              variant="body1"
            >
              Apply to All Channel IDs
            </WithStyles(ForwardRef(Typography))>
          </Styled(MuiBox)>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <br />
    <WithStyles(ForwardRef(Grid))
      container={true}
      justify="flex-start"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        sm={12}
      >
        <label
          className="makeStyles-label-2"
          testId="SelectToolLabel"
        >
          Select third party tool for Automatic Detection
        </label>
        <Styled(MuiBox)
          mb={2}
          pt={1}
        >
          <WithStyles(ForwardRef(FormControl))
            fullWidth={true}
            size="small"
          >
            <WithStyles(ForwardRef(Select))
              MenuProps={
                Object {
                  "anchorOrigin": Object {
                    "horizontal": "left",
                    "vertical": "bottom",
                  },
                  "getContentAnchorEl": null,
                }
              }
              id="selectTool"
              name="toolForAutomaticDetection"
              onChange={[Function]}
              value="tisane"
              variant="outlined"
            >
              <WithStyles(ForwardRef(MenuItem))
                value="tisane"
              >
                Tisane.ai
              </WithStyles(ForwardRef(MenuItem))>
              <WithStyles(ForwardRef(MenuItem))
                value="siftninja"
              >
                SiftNinja
              </WithStyles(ForwardRef(MenuItem))>
            </WithStyles(ForwardRef(Select))>
          </WithStyles(ForwardRef(FormControl))>
        </Styled(MuiBox)>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <br />
    <Styled(MuiBox)
      pt={1}
    >
      <WithStyles(ForwardRef(Grid))
        container={true}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <label
            className="makeStyles-label-2"
            testid="tisaneApiKey"
          >
            Tisane.ai API Key
          </label>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <Styled(MuiBox)
            className="makeStyles-infoIcon-9"
          >
            <WithStyles(WithStyles(ForwardRef(Tooltip)))
              interactive={true}
              title={
                <React.Fragment>
                  <span>
                    Sign up with 
                  </span>
                  <a
                    href="https://tisane.ai/signup/"
                    style={
                      Object {
                        "color": "black",
                        "fontWeight": 600,
                        "textDecorationLine": "none",
                      }
                    }
                  >
                    Tisane.ai
                  </a>
                  <span>
                     locate your key under your 
                  </span>
                  <a
                    href="https://tisane.ai/developer/"
                    style={
                      Object {
                        "color": "black",
                        "fontWeight": 600,
                        "textDecorationLine": "none",
                      }
                    }
                  >
                    Developer Profile.
                  </a>
                  <span>
                     Use primary or secondary key. 
                  </span>
                </React.Fragment>
              }
            >
              <img
                alt="info-circle"
                src="/images/info-circle.svg"
              />
            </WithStyles(WithStyles(ForwardRef(Tooltip)))>
          </Styled(MuiBox)>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
      <Styled(MuiBox)
        mb={2}
        pt={1}
      >
        <WithStyles(ForwardRef(TextField))
          fullWidth={true}
          id="tisaneApiKey"
          name="tisaneApiKey"
          onChange={[Function]}
          placeholder="Tisane.ai API key"
          size="small"
          value=""
          variant="outlined"
        />
      </Styled(MuiBox)>
    </Styled(MuiBox)>
    <br />
    <WithStyles(ForwardRef(Grid))
      container={true}
      justify="flex-start"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        sm={5}
      >
        <label
          testid="lang"
        >
          Language
        </label>
        <Styled(MuiBox)
          mb={2}
          pt={1}
        >
          <WithStyles(ForwardRef(FormControl))
            fullWidth={true}
            placeholder="Langauge"
            size="small"
          >
            <WithStyles(ForwardRef(Select))
              MenuProps={
                Object {
                  "anchorOrigin": Object {
                    "horizontal": "left",
                    "vertical": "bottom",
                  },
                  "getContentAnchorEl": null,
                }
              }
              id="lang"
              inputProps={
                Object {
                  "aria-label": "Without label",
                }
              }
              name="tisaneLanguage"
              onChange={[Function]}
              value="Hindi"
              variant="outlined"
            >
              <WithStyles(ForwardRef(MenuItem))
                key="0"
                value="Autodetect"
              >
                Autodetect
              </WithStyles(ForwardRef(MenuItem))>
              <WithStyles(ForwardRef(MenuItem))
                key="1"
                value="English"
              >
                English
              </WithStyles(ForwardRef(MenuItem))>
              <WithStyles(ForwardRef(MenuItem))
                key="2"
                value="Spanish"
              >
                Spanish
              </WithStyles(ForwardRef(MenuItem))>
              <WithStyles(ForwardRef(MenuItem))
                key="3"
                value="Portugese"
              >
                Portugese
              </WithStyles(ForwardRef(MenuItem))>
              <WithStyles(ForwardRef(MenuItem))
                key="4"
                value="French"
              >
                French
              </WithStyles(ForwardRef(MenuItem))>
            </WithStyles(ForwardRef(Select))>
          </WithStyles(ForwardRef(FormControl))>
        </Styled(MuiBox)>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <br />
    <WithStyles(ForwardRef(Grid))
      container={true}
      justify="space-between"
      spacing={6}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        lg={6}
        md={6}
        sm={6}
        xs={12}
      >
        <Styled(MuiBox)
          pl={1}
        >
          <label
            className="makeStyles-label-2"
            testid="BigotryLevel"
          >
            Filter Level For Bigotry (Hate Speech)
          </label>
          <Styled(MuiBox)
            pt={1}
          >
            <SliderComponent
              id="BigotryLevel"
              name="tisane"
              onChange={[Function]}
              value={0.5}
            />
          </Styled(MuiBox)>
        </Styled(MuiBox)>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        lg={6}
        md={6}
        sm={6}
        xs={6}
      >
        <Styled(MuiBox)
          pl={1}
        >
          <label
            className="makeStyles-label-2"
            testid="CyberBullying"
          >
            Filter Level For Personal Attacks (Cyberbullying)
          </label>
          <Styled(MuiBox)
            pt={1}
          >
            <SliderComponent
              id="CyberBullying"
              name="tisane"
              onChange={[Function]}
              value={0.5}
            />
          </Styled(MuiBox)>
        </Styled(MuiBox)>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        lg={6}
        md={6}
        sm={6}
        xs={12}
      >
        <Styled(MuiBox)
          pl={1}
        >
          <label
            className="makeStyles-label-2"
            testid="criminalLevel"
          >
            Filter Level For Criminal Activity
          </label>
          <Styled(MuiBox)
            pt={1}
          >
            <SliderComponent
              id="criminalLevel"
              name="tisane"
              onChange={[Function]}
              value={0.5}
            />
          </Styled(MuiBox)>
        </Styled(MuiBox)>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        lg={6}
        md={6}
        sm={6}
        xs={6}
      >
        <Styled(MuiBox)
          pl={1}
        >
          <label
            className="makeStyles-label-2"
            testid="sexualLevel"
          >
            Filter Level For Sexual Advances
          </label>
          <Styled(MuiBox)
            pt={1}
          >
            <SliderComponent
              id="sexualLevel"
              name="tisane"
              onChange={[Function]}
              value={0.5}
            />
          </Styled(MuiBox)>
        </Styled(MuiBox)>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        lg={6}
        md={6}
        sm={6}
        xs={12}
      >
        <Styled(MuiBox)
          pl={1}
        >
          <label
            className="makeStyles-label-2"
            testid="ProfanityLevel"
          >
            Filter Level For Profanity
          </label>
          <Styled(MuiBox)
            pt={1}
          >
            <SliderComponent
              id="ProfanityLevel"
              name="tisane"
              onChange={[Function]}
              value={0.5}
            />
          </Styled(MuiBox)>
        </Styled(MuiBox)>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      justify="flex-start"
      spacing={6}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        md={6}
        sm={6}
      >
        <label
          className="makeStyles-label-2"
          testid="blockOrMask"
        >
          When profanity is detected
        </label>
        <Styled(MuiBox)
          mb={2}
          pt={1}
        >
          <WithStyles(ForwardRef(FormControl))
            fullWidth={true}
            size="small"
          >
            <WithStyles(ForwardRef(Select))
              MenuProps={
                Object {
                  "anchorOrigin": Object {
                    "horizontal": "left",
                    "vertical": "bottom",
                  },
                  "getContentAnchorEl": null,
                }
              }
              id="blockOrMask"
              inputProps={
                Object {
                  "aria-label": "Without label",
                }
              }
              name="automaticDetectionModType"
              onChange={[Function]}
              value="mask-message"
              variant="outlined"
            >
              <WithStyles(ForwardRef(MenuItem))
                value="mask-message"
              >
                Mask Message
              </WithStyles(ForwardRef(MenuItem))>
              <WithStyles(ForwardRef(MenuItem))
                value="block-message"
              >
                Block Message
              </WithStyles(ForwardRef(MenuItem))>
            </WithStyles(ForwardRef(Select))>
          </WithStyles(ForwardRef(FormControl))>
        </Styled(MuiBox)>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        md={6}
        sm={6}
      >
        <WithStyles(ForwardRef(Grid))
          container={true}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            <label
              className="makeStyles-label-2"
              testid="setChar"
            >
              Set a masking character
            </label>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            <Styled(MuiBox)
              className="makeStyles-infoIcon-9"
            >
              <WithStyles(WithStyles(ForwardRef(Tooltip)))
                title="Note: Default value will be *"
              >
                <img
                  alt="info-circle"
                  src="/images/info-circle.svg"
                />
              </WithStyles(WithStyles(ForwardRef(Tooltip)))>
            </Styled(MuiBox)>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
        <Styled(MuiBox)
          pt={1}
        >
          <WithStyles(ForwardRef(TextField))
            error={false}
            fullWidth={true}
            id="setChar"
            inputProps={
              Object {
                "maxLength": 1,
              }
            }
            name="automaticDetectionCharacterToMaskWith"
            onChange={[Function]}
            placeholder="Set"
            size="small"
            value="*"
            variant="outlined"
          />
        </Styled(MuiBox)>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <br />
    <WithStyles(ForwardRef(Grid))
      container={true}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(ForwardRef(Grid))
          container={true}
          justify="flex-start"
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            <WithStyles(ForwardRef(Checkbox))
              checked={false}
              id="reRoute"
              name="automaticDetectionReRouteMessages"
              onChange={[Function]}
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            <Styled(MuiBox)
              pl={1}
              pt={1}
            >
              <WithStyles(ForwardRef(Typography))
                testid="reRouteMsg"
                variant="body1"
              >
                Route messages to
              </WithStyles(ForwardRef(Typography))>
            </Styled(MuiBox)>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <Styled(MuiBox)>
          <WithStyles(ForwardRef(Chip))
            className="makeStyles-reroute-8"
            label="banned.*"
          />
        </Styled(MuiBox)>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <br />
    <WithStyles(ForwardRef(Grid))
      container={true}
      justify="space-between"
      spacing={3}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      />
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(ForwardRef(Button))
          className="makeStyles-saveButtonStyle-5"
          disabled={false}
          id="save"
          variant="contained"
        >
          Save
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Card))>
</Fragment>
`;
